sections/design/why-react.tex

```diff
@@ -0,0 +1,12 @@
+\section{Why React}
+For our frontend, we decided to use React.
+React is a JavaScript library for creating web applications\cite{ReactJS}. 
+\\
+React is one of the most popular choices when it comes to frontend development\cite{NPMVueReactAngular}.
+It is created by Facebook and used for their website as well as numerous other websites\cite{ReactHistory}.
+React is lightweight and allows the user to choose what additional libraries they would want to use in their project, which keeps its size down.
+\\
+It also has an extensive ecosystem that offers various react specific libraries that one could use in their web applications.
+Being this popular results in a very robust ecosystem with lots of users, but this is also needed as the additional libraries are maintained exclusively by the react community and are not a part of the core React library, unlike other frontend libraries such as Vue\cite{Vue}.
+\\\\
+React was created so that it would be easier to make reusable UI components that can handle data that changes regularly\cite{ReactHistory}.
```

Comment:
```suggestion
React was created so that it would be easier to make reusable UI components that can handle data that changes regularly \cite{ReactHistory}.
```